healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageList.java

```diff
@@ -63,21 +59,21 @@ public static void hl7v2MessageList(String hl7v2StoreName) throws IOException {
               .setPageSize(100) // Specify pageSize up to 1000
               .setPageToken(pageToken);
 
+      ListMessagesResponse response;
       // Execute response and collect results.
-      ListMessagesResponse response = request.execute();
-      Collection<Message> responseMessages = response.getHl7V2Messages();
-      if (responseMessages != null) {
-        messages.addAll(responseMessages);
-      }
-
-      // Update the page token for the next request.
-      pageToken = response.getNextPageToken();
-    } while (pageToken != null);
-
-    // Print results.
-    System.out.printf("Retrieved %s HL7v2 messages: \n", messages.size());
-    for (Message data : messages) {
-      System.out.println("\t" + data.getData());
+      do {
+        response = request.execute();
+        if (response.getHl7V2Messages() == null) {
+          continue;
+        }
+        System.out.printf("Retrieved %s HL7v2 messages: \n", response.getHl7V2Messages().size());
+        for (Message message : response.getHl7V2Messages()) {
+            System.out.println("\t" + message);
+        }
+        request.setPageToken(response.getNextPageToken());
+       } while (response.getNextPageToken() != null);
+    } catch (IOException e) {
+      System.out.println("Unable to list HL7v2 messages: \n" + e.toString());
```

Comment:
You don't need the extra `\n` when using println

Reply:
Done.